pkg/sql/schema_resolver.go

```diff
@@ -528,6 +528,7 @@ func maybeLookUpUDF(
 	}
 
 	var udfDef *tree.ResolvedFunctionDefinition
+	var schemaLookedup catalog.DescriptorIDSet
```

Comment:
I think this is going to hurt performance in the happy path. I wonder if the `SearchPath` object itself should track duplicates. I think it'd be cheaper generally if we put a `FastIntSet` in the `SearchPath` and then had a way to ask if a schema was duplicated. WDYT?

Reply:
I think it'd be okay for that to be quadratic.

Consider:
```go
diff --git a/pkg/sql/sessiondata/search_path.go b/pkg/sql/sessiondata/search_path.go
index 8d92fc799b0..018f6fa3b86 100644
--- a/pkg/sql/sessiondata/search_path.go
+++ b/pkg/sql/sessiondata/search_path.go
@@ -19,6 +19,7 @@ import (
        "github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
        "github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
        "github.com/cockroachdb/cockroach/pkg/sql/sem/catconstants"
+       "github.com/cockroachdb/cockroach/pkg/util/intsets"
 )
 
 // DefaultSearchPath is the search path used by virgin sessions.
@@ -35,6 +36,7 @@ type SearchPath struct {
        containsPgTempSchema bool
        tempSchemaName       string
        userSchemaName       string
+       duplicatedIndexes    intsets.Fast
 }
 
 // EmptySearchPath is a SearchPath with no schema names in it.
@@ -49,25 +51,29 @@ func DefaultSearchPathForUser(userName username.SQLUsername) SearchPath {
 // MakeSearchPath returns a new immutable SearchPath struct. The paths slice
 // must not be modified after hand-off to MakeSearchPath.
 func MakeSearchPath(paths []string) SearchPath {
-       containsPgCatalog := false
-       containsPgExtension := false
-       containsPgTempSchema := false
-       for _, e := range paths {
+       exists := func(haystack []string, needle string) bool {
+               for i := range haystack {
+                       if haystack[i] == needle {
+                               return true
+                       }
+               }
+               return false
+       }
+       sp := SearchPath{paths: paths}
+       for i, e := range paths {
                switch e {
                case catconstants.PgCatalogName:
-                       containsPgCatalog = true
+                       sp.containsPgCatalog = true
                case catconstants.PgTempSchemaName:
-                       containsPgTempSchema = true
+                       sp.containsPgTempSchema = true
                case catconstants.PgExtensionSchemaName:
-                       containsPgExtension = true
+                       sp.containsPgExtension = true
+               }
+               if exists(paths[:i], e) {
+                       sp.duplicatedIndexes.Add(i)
                }
        }
-       return SearchPath{
-               paths:                paths,
-               containsPgCatalog:    containsPgCatalog,
-               containsPgExtension:  containsPgExtension,
-               containsPgTempSchema: containsPgTempSchema,
-       }
+       return sp
 }
 
```

Reply:
Fantastic idea. I like it.

Reply:
Done what you suggested!